tests/snappi_tests/multidut/pfc/files/m2o_oversubscribe_lossy_helper.py

```diff
@@ -0,0 +1,432 @@
+from tests.common.snappi_tests.snappi_helpers import wait_for_arp
+from tests.common.snappi_tests.snappi_test_params import SnappiTestParams
+
+logger = logging.getLogger(__name__)
+
+PAUSE_FLOW_NAME = 'Pause Storm'
+TEST_FLOW_NAME = 'Test Flow'
+TEST_FLOW_AGGR_RATE_PERCENT = 30
+BG_FLOW_NAME = 'Background Flow'
+BG_FLOW_AGGR_RATE_PERCENT = 25
+DATA_PKT_SIZE = 1024
+DATA_FLOW_DURATION_SEC = 2
+DATA_FLOW_DELAY_SEC = 1
+SNAPPI_POLL_DELAY_SEC = 2
+TOLERANCE_THRESHOLD = 0.05
+
+
+def run_pfc_m2o_oversubscribe_lossy_test(api,
+                                         testbed_config,
+                                         port_config_list,
+                                         conn_data,
+                                         fanout_data,
+                                         dut_port,
+                                         pause_prio_list,
+                                         test_prio_list,
+                                         bg_prio_list,
+                                         prio_dscp_map,
+                                         snappi_extra_params=None):
+    """
+    Run PFC watchdog test in a multi-node (>=3) topoology
+
+    Args:
+        api (obj): SNAPPI session
+        testbed_config (obj): testbed L1/L2/L3 configuration
+        port_config_list (list): list of port configuration
+        conn_data (dict): the dictionary returned by conn_graph_fact.
+        fanout_data (dict): the dictionary returned by fanout_graph_fact.
+        duthost (Ansible host instance): device under test
+        dut_port (str): DUT port to test
+        pause_prio_list (list): priorities to pause for PFC pause storm
+        test_prio_list (list): priorities of test flows
+        bg_prio_list (list): priorities of background flows
+        prio_dscp_map (dict): Priority vs. DSCP map (key = priority)
+        snappi_extra_params (SnappiTestParams obj): additional parameters for Snappi traffic
+
+    Returns:
+        N/A
+    """
+    if snappi_extra_params is None:
+        snappi_extra_params = SnappiTestParams()
+
+    duthost1 = snappi_extra_params.multi_dut_params.duthost1
+    rx_port = snappi_extra_params.multi_dut_params.multi_dut_ports[0]
+    rx_port_id_list = [rx_port["port_id"]]
+    duthost2 = snappi_extra_params.multi_dut_params.duthost2
+    tx_port = [snappi_extra_params.multi_dut_params.multi_dut_ports[1],
+               snappi_extra_params.multi_dut_params.multi_dut_ports[2]]
+    tx_port_id_list = [tx_port[0]["port_id"], tx_port[1]["port_id"]]
+
+    pytest_assert(testbed_config is not None, 'Fail to get L2/3 testbed config')
+    stop_pfcwd(duthost1, rx_port['asic_value'])
+    disable_packet_aging(duthost1)
+    stop_pfcwd(duthost2, tx_port[0]['asic_value'])
+    disable_packet_aging(duthost2)
+
+    test_flow_rate_percent = int(TEST_FLOW_AGGR_RATE_PERCENT)
+    bg_flow_rate_percent = int(BG_FLOW_AGGR_RATE_PERCENT)
+    exp_dur_sec = 5
+    __gen_traffic(testbed_config=testbed_config,
```

Comment:
why does this test not use functions in `traffic_generator.py`? The logic appears identical.